src/main/kotlin/net/atos/zac/zoeken/IndexingService.kt

```diff
@@ -0,0 +1,303 @@
+/*
+ * SPDX-FileCopyrightText: 2022 Atos
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package net.atos.zac.zoeken
+
+import jakarta.enterprise.inject.Instance
+import jakarta.inject.Inject
+import jakarta.inject.Singleton
+import net.atos.client.zgw.drc.DrcClientService
+import net.atos.client.zgw.drc.model.EnkelvoudigInformatieobjectListParameters
+import net.atos.client.zgw.shared.ZGWApiService
+import net.atos.client.zgw.shared.model.Results
+import net.atos.client.zgw.shared.util.URIUtil
+import net.atos.client.zgw.zrc.ZrcClientService
+import net.atos.client.zgw.zrc.model.ZaakListParameters
+import net.atos.zac.app.task.model.TaakSortering
+import net.atos.zac.flowable.task.FlowableTaskService
+import net.atos.zac.shared.model.SorteerRichting
+import net.atos.zac.util.UriUtil
+import net.atos.zac.zoeken.converter.AbstractZoekObjectConverter
+import net.atos.zac.zoeken.model.ZoekObject
+import net.atos.zac.zoeken.model.index.ZoekObjectType
+import nl.lifely.zac.util.AllOpen
+import org.apache.solr.client.solrj.SolrClient
+import org.apache.solr.client.solrj.SolrQuery
+import org.apache.solr.client.solrj.SolrServerException
+import org.apache.solr.client.solrj.impl.Http2SolrClient
+import org.apache.solr.client.solrj.response.QueryResponse
+import org.apache.solr.common.params.CursorMarkParams
+import org.eclipse.microprofile.config.ConfigProvider
+import java.io.IOException
+import java.util.UUID
+import java.util.logging.Logger
+
+@Singleton
+@AllOpen
+@Suppress("TooManyFunctions")
+class IndexingService @Inject constructor(
+    private val converterInstances: Instance<AbstractZoekObjectConverter<out ZoekObject>>,
+    private val zrcClientService: ZrcClientService,
+    private val drcClientService: DrcClientService,
+    private val flowableTaskService: FlowableTaskService
+) {
+    companion object {
+        const val SOLR_CORE = "zac"
+
+        private const val SOLR_MAX_RESULT = 100
+        private const val TAKEN_MAX_RESULTS = 50
+
+        private val LOG = Logger.getLogger(IndexingService::class.java.name)
+        private val reindexingViewfinder = mutableSetOf<ZoekObjectType>()
```

Comment:
so I think a plain hashmap is not thread safe. Better to use a Java ConcurrentHashMap here I think or maybe an Kotlin equivalent (something Atomic). Maybe we will never run into such concurrency issues but I think better be safe here.